examples/s3/bucketaclcanned.yaml

```diff
@@ -0,0 +1,23 @@
+apiVersion: s3.aws.upbound.io/v1beta1
```

Comment:
When we start triggering automated tests with uptest (e.g., by testing the `examples/s3/bucketacl.yaml` file), this example will not be covered. We can also reconsider how to trigger the tests automatically but it simplifies to rely on the examples/\<group\>/\<resource\>.yaml convention. 

We don't need to decide on how to improve this in this PR, but how about moving this example into the `examples/s3/bucketacl.yaml` file if they can share common dependencies? Or maybe better, we can have another convention that will help us detecting what to test. A proposal would be to name the file with something like `bucketacl-canned.yaml`. The reasoning is `bucketaclcanned` could be another resource's name but we can easily deduce the test target resource name with this convention. We could as well just the file name as `bucketaclcanned.yaml` but in our tooling inspect the resource manifests in the file contents and deduce the test target resource (we were using a specific test annotation to mark the test target previously, or we can just assume the first manifest is the test target manifest). This way we would not need any conventions for the file names. I lean towards a convention like `bucketacl-canned.yaml` as it looks simpler to me. What do you think? cc. @sergenyalcin

Reply:
@ulucinar that makes sense, I can rename the file. Do we actually have a code for mass example execution with uptest? Currently I am aware of targeted per example one only

Reply:
Yes, group-wide test executions (like specifying `examples/ec2` as the test target) have proven to be problematic. So, we are triggering fine-granular, specific resource-targeted tests (like `examples/s3/bucketacl.yaml`) as you mentioned. Even if we adopt the naming convention suggested above, we would still need changes in our tooling (either in upjet itself, or in the clients of upjet) so that when we intent to test the `s3.BucketACL` managed resource implementation, we would use manifests from `bucketacl.yaml` and `bucketacl-*.yaml` (this is not supported yet). 

The idea is to have all the use cases captured in the example manifests to be covered by our automated tests. The way to do this is right now putting all examples into a single file (e.g., `examples/s3/bucketacl.yaml`), excluding the directory-targeted tests. But this does not scale well in my opinion, and having per use-case example manifests looks better to me. This is something we need to consider when we continue with the triggering tests automatically work stream (epic).